picard/ui/mainwindow.py

```diff
@@ -903,31 +904,39 @@ def _update_cd_lookup_actions(self, result=None, error=None):
             self.update_cd_lookup_drives(result)
 
     def update_cd_lookup_drives(self, drives):
+        self.cd_lookup_menu.clear()
+        self.cd_lookup_action.setEnabled(discid is not None)
         if not drives:
             log.warning("CDROM: No CD-ROM drives found - Lookup CD functionality disabled")
+            has_drives = False
         else:
+            has_drives = True
             config = get_config()
             shortcut_drive = config.setting["cd_lookup_device"].split(",")[0] if len(drives) > 1 else ""
-            self.cd_lookup_action.setEnabled(discid is not None)
-            self.cd_lookup_menu.clear()
             for drive in drives:
                 action = self.cd_lookup_menu.addAction(drive)
                 action.setData(drive)
                 if drive == shortcut_drive:
-                    # Clear existing shortcode on main action and assign it to sub-action
-                    self.cd_lookup_action.setShortcut(QtGui.QKeySequence())
-                    action.setShortcut(QtGui.QKeySequence(_("Ctrl+K")))
```

Comment:
What happens to this shortcut?

Reply:
It stays on `self.cd_lookup_action` , which then just triggers the default action from the menu. That's way easier then to change the shortcut only. But I also needed this so it triggers the eaclog action properly.